sourcegraph/deploy-sourcegraph
# add back language extensions to deploy-sourcegraph

TODO: 

- [x] Add example TLS configurations for Go, Typescript
   - [x] ask @chrismwendt if it's safe to not use https if they're not exposed over the internet
   - [x] add example Ingress configurations
   - [ ] ~add example TLS_CERT, TLS_KEY configurations~ -> not handling non- nginx-ingress deployments for now
   - [x] write docs to show how to use http basic auth with nginx-ingress

- [x] write docs for configure.md
- [x] figure out whether or not resource limits/requests are reasonable 
- [ ] switch docker tags to non-`insiders` ones


---

Test plan: 

1. Spin up a new test cluster with https://github.com/sourcegraph/deploy-k8s-helper/tree/lang-test and `deploySourcegraphRoot` checked out to https://github.com/sourcegraph/deploy-sourcegraph/tree/add-back-lang-ext

1. Use mkcert to generate a certificate for sourcegraph.example.com

   ```console
   > mkcert  -cert-file cert -key-file key 35.232.68.120
   ```

1. Run `cat ./cert | pulumi config --secret set certificate` and `cat ./cert | pulumi config --secret set key`

1. Create htpassword using instructions in https://github.com/sourcegraph/deploy-sourcegraph/blob/add-back-lang-ext/configure/lang/go/README.md

    ```console
    > htpasswd -c auth langserveruser 
    ```

1.  Run `cat ./auth | pulumi config --secret set htpassword`

1. Run `yarn up`

1. Add the ip address from ingressIPs to `/etc/hosts`

1. Add an external service sourcegraph.example.com, clone sourcegraph/sourcegraph on it 

1. Use Global settings from https://github.com/sourcegraph/deploy-sourcegraph/blob/add-back-lang-ext/configure/lang/go/README.md and https://github.com/sourcegraph/deploy-sourcegraph/blob/add-back-lang-ext/configure/lang/typescript/README.md to connect to the language servers

1. Open a Go and Typescript file from sourcegraph/sourcegraph. You should see code intelligence, and your network tab should show succesfull communication to the language servers
